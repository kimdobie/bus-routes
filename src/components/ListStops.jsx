import React from 'react';
import PropTypes from 'prop-types';

const axios = require('axios');
// import Form from 'react-bootstrap/Form';

class ListStops extends React.Component {
  constructor(props) {
    super(props);
    // const { route } = this.props;
    this.state = {
      stops: [] // array of route objects
    };
    this.setListItems = this.setListItems.bind(this);
    this.getStops = this.getStops.bind(this);
    this.getStops();
  }

  getStops() {
    const { route, direction } = this.props;
    if (route === '' || direction === '') {
      this.setState({ stops: [] });
      return;
    }
    axios
      .get(`/api/Stops/${route}/${direction}`)
      .then(routes => {
        this.setState({ stops: routes.data });
      })
      .catch(error => {
        // eslint-disable-next-line no-console
        console.log('There was an issue getting stops: ', error);
      });
  }

  setListItems() {
    const { stops } = this.state;
    return stops.map(stopObj => {
      return <li key={stopObj.Value}>{stopObj.Text}</li>;
    });
  }

  render() {
    // const { stops } = this.state;
    return (
      <div>
        <h2>Stop:</h2>
        <ul>{this.setListItems()}</ul>
      </div>
    );
  }
}
export default ListStops;

ListStops.propTypes = {
  route: PropTypes.string,
  direction: PropTypes.string
};
ListStops.defaultProps = {
  route: '',
  direction: ''
};
