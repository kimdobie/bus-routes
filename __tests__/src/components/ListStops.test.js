import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ListStops from '../../../src/components/ListStops';

const fs = require('fs');
const path = require('path');
const axios = require('axios');

configure({ adapter: new Adapter() });

jest.mock('axios');

describe('App tests', () => {
  const routes = fs.readFileSync(
    path.join(__dirname, '../../../__fixtures__/routes.json')
  );
  const directions = fs.readFileSync(
    path.join(__dirname, '../../../__fixtures__/directions.json')
  );
  const stops = fs.readFileSync(
    path.join(__dirname, '../../../__fixtures__/stops.json')
  );
  axios.setRoutes(JSON.parse(routes));
  axios.setDirections(JSON.parse(directions));
  axios.setStops(JSON.parse(stops));

  test('Snapshot test without routes or directions', async () => {
    // eslint-disable-next-line react/jsx-filename-extension
    const listStops = await shallow(<ListStops />);
    expect(listStops.state().stops).toEqual([]);
    expect(listStops.find('li')).toHaveLength(0);
    expect(listStops).toMatchSnapshot();
  });

  test('Snapshot test with routes and directions', async () => {
    const listStops = await shallow(<ListStops route='123' direction='456' />);

    expect(listStops.state().stops).not.toEqual([]);
    const numStops = JSON.parse(stops).length;
    expect(listStops.find('li')).toHaveLength(numStops);

    expect(listStops).toMatchSnapshot();
  });
});
