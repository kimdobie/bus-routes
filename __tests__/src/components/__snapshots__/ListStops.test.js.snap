// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Snapshot test with routes and directions 1`] = `
<div>
  <h2>
    Stop:
  </h2>
  <ul>
    <li
      key="MAAM"
    >
      Mall of America Transit Station
    </li>
    <li
      key="77PO"
    >
      Portland Ave and 77th St
    </li>
    <li
      key="66PO"
    >
      Portland Ave and 66th St
    </li>
    <li
      key="56CH"
    >
      Chicago Ave and 56th St
    </li>
    <li
      key="46CH"
    >
      Chicago Ave and 46th St
    </li>
    <li
      key="38CH"
    >
      Chicago Ave and 38th St
    </li>
    <li
      key="CHLA"
    >
      Chicago Lake Transit Center
    </li>
    <li
      key="8SCH"
    >
      Chicago Ave and 8th St 
    </li>
    <li
      key="7SNI"
    >
      7th St  and Nicollet Mall
    </li>
    <li
      key="7SOL"
    >
      7th St  and Olson Memorial Hwy
    </li>
    <li
      key="EMBD"
    >
      Emerson Ave  and W Broadway Ave
    </li>
    <li
      key="26PN"
    >
      26th Ave and Penn Ave
    </li>
    <li
      key="33FM"
    >
      33rd Ave  and Fremont Ave 
    </li>
    <li
      key="44FM"
    >
      44th Ave  and Fremont Ave 
    </li>
    <li
      key="47OS"
    >
      Osseo Rd and 47th Ave 
    </li>
    <li
      key="BCTC"
    >
      Brooklyn Center Transit Center
    </li>
  </ul>
</div>
`;

exports[`App tests Snapshot test without routes or directions 1`] = `
<div>
  <h2>
    Stop:
  </h2>
  <ul />
</div>
`;
